src/middleware.ts

```diff
@@ -191,7 +191,18 @@ export const persist = <S extends State>(
     // prevent error if the storage is not defined (e.g. when server side rendering a page)
   }
 
-  if (!storage) return config(set, get, api)
+  if (!storage) {
+    return config(
+      (...args) => {
+        console.warn(
+          `Persist middleware: unable to update ${name}, the given storage is currently unavailable.`
+        )
+        set(...args)
+      },
+      get,
+      api
+    )
+  }
```

Comment:
Does it need to warn all the time when updating? why not just at the config time?
```js
  if (!storage) {
    console.warn(...)
    return config(get, set, api)
  }
```

Reply:
As I said in #283, it'll probably be annoying for the user to be spamed if he has a few stores. Especially if he doesn't especially use the store(s) while the storage is unavailable.

Reply:
I see, but this would be annoying too.
Anyway, if it's intentional, I'm ok with it.

Reply:
Well, I guess this warning is always going to be annoying wherever we put it. The question is where it's going to be just annoying enough for the user to notice it without spending hours to debug why the data is not in the storage :smile: